### PR TITLE
Implement note chat

### PR DESCRIPTION
**Pull request by kristenbrann**
Screen Shot 2023-03-20 at 5 38 32 PM

And on hover there is a copy button to copy messages
Screen Shot 2023-03-20 at 7 13 31 PM

That way you could copy something like this into your document
Screen_Shot_2023-03-20_at_5 39 46_PM

### CODE REVIEW

**Comment by cpaika** [comment]
meta comment - can you start developing on branches in this repo?  Easier for me to check out your branches locally for playing around

File: src/main.ts [hunk 1]
```diff
@@ -192,9 +215,25 @@ export default class VaultChat extends Plugin {
...
 		}))
 		return results.filter(isSearchResult)
 	}
+
+	showNoteChatView() {
+		this.app.workspace.detachLeavesOfType(NOTE_CHAT_VIEW);
+		this.app.workspace.getRightLeaf(false).setViewState({
+			type: NOTE_CHAT_VIEW,
+			active: true,
+		}).then(() => {
```
Comment (cpaika) [inline src/main.ts:230]: What is this doing here?

File: src/components/ChatGPTConversation.tsx [hunk 2]
```diff
@@ -3,18 +3,22 @@ import { ChatCompletionRequestMessage } from "openai/api";
...
...1009,0.0469999 h3.9682999c1.6041985,0,2.9096985,1.2609005,2.9096985,2.8115005V59.1567993z"></path> <path d="M38.6031494,13.2063999H16.253952c-0.5615005,0-1.0159006,0.4542999-1.0159006,1.0158005 c0,0.5615997,0.4544001,1.0158997,1.0159006,1.0158997h22.3491974c0.5615005,0,1.0158997-0.4542999,1.0158997-1.0158997 C39.6190491,13.6606998,39.16465,13.2063999,38.6031494,13.2063999z"></path> <path d="M38.6031494,21.3334007H16.253952c-0.5615005,0-1.0159006,0.4542999-1.0159006,1.0157986 c0,0.5615005,0.4544001,1.0159016,1.0159006,1.0159016h22.3491974c0.5615005,0,1.0158997-0.454401,1.0158997-1.0159016 C39.6190491,21.7877007,39.16465,21.3334007,38.6031494,21.3334007z"></path> <path d="M38.6031494,29.4603004H16.253952c-0.5615005,0-1.0159006,0.4543991-1.0159006,1.0158997 s0.4544001,1.0158997,1.0159006,1.0158997h22.3491974c0.5615005,0,1.0158997-0.4543991,1.0158997-1.0158997 S39.16465,29.4603004,38.6031494,29.4603004z"></path> <path d="M28.4444485,37.5872993H16.253952c-0.5615005,0-1.0159006,0.4543991-1.0159006,1.0158997 s0.4544001,1.0158997,1.0159006,1.0158997h12.1904964c0.5615025,0,1.0158005-0.4543991,1.0158005-1.0158997 S29.0059509,37.5872993,28.4444485,37.5872993z"></path> </g> </g></svg></button>
```
Comment (cpaika) [inline src/components/ChatGPTConversation.tsx:19]: What is this?

**Comment by kristenbrann** [comment]
> meta comment - can you start developing on branches in this repo? Easier for me to check out your branches locally for playing around

Yes of course! Sorry about that!

File: src/components/ChatGPTConversation.tsx (hunk 2 above)
Comment (kristenbrann) [reply src/components/ChatGPTConversation.tsx:19]: SVG for the copy icon

**Review by cpaika — APPROVED** [review]
LGTM!

File: src/main.ts (hunk 1 above)
Comment (kristenbrann) [reply src/main.ts:230]: This is how you display the view per the docs: https://marcus.se.net/obsidian-plugin-docs/user-interface/views

**Pull request merged**